modules/ServerAPI/src/server/CustomerSpecificServerFunctions.coffee

```diff
@@ -163,19 +163,25 @@ getSystemRolesFromSSOProfile = (profile) =>
 	return roles
 
 exports.ssoLoginStrategy = (req, profile, callback) ->
+	exports.logUsage "login attempt", JSON.stringify(ip: req.ip, referer: req.headers['referer'], agent: req.headers['user-agent']), JSON.stringify(profile)
 	config = require '../../../conf/compiled/conf.js'
 	serverUtilityFunctions = require "#{ACAS_HOME}/routes/ServerUtilityFunctions.js"
 	authorRoutes = require '../../../routes/AuthorRoutes.js'
 
 	userNameAttribute = config.all.server.security.saml.userNameAttribute
-	console.log("Incoming login #{JSON.stringify(profile)}")
 
+	# Expected profile keys
+	expectedKeys = [config.all.server.security.saml.userNameAttribute, config.all.server.security.saml.firstNameAttribute, config.all.server.security.saml.lastNameAttribute, config.all.server.security.saml.emailAttribute]
+	missingFromProfile = expectedKeys.filter (value) -> !Object.keys(profile).includes(value)
+	if missingFromProfile.length > 0
+		console.error "Missing expected key(s) from returned user profile #{JSON.stringify(missingFromProfile)}"
+		return callback null, false, message: "Configured profile attributes are different than those configured on the IDP"
```

Comment:
Any reason not to include the keys in the error message and only in the ACAS logs?

Reply:
Was just trying to not be overkill with the messages to the user but I will add it because it will be helpful.